### PR TITLE
[ProcessGroupNCCL] Move new trace utils

### PR DESCRIPTION
**Pull request by kwen2501**
to TraceUtils.h

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @kiukchung @d4l3k @lucasllc

### CODE REVIEW

File: torch/csrc/distributed/c10d/TraceUtils.h [hunk 1]
```diff
@@ -256,4 +258,260 @@ inline std::string retrieveDesyncReport(
   return report;
 }
 
+/* Trace Utils Related to Flight Recorder */
```
Comment (wconstab) [inline torch/csrc/distributed/c10d/TraceUtils.h:261]: maybe also comment that this is only used by PGNCCL (could be generalized in an ideal world but wasn't done that way, so isn't expected to be fully general at the moment)

**Review by wconstab — APPROVED** [review]
i think this is fine. cc @zdevito @fduwjj

File: torch/csrc/distributed/c10d/TraceUtils.h (hunk 1 above)
Comment (kwen2501) [reply torch/csrc/distributed/c10d/TraceUtils.h:264]: Thanks, done!

**Review by XilunWu — APPROVED** [review]
LGTM!

**Comment by ChuBoning** [comment]
what about third-party devices？Many symbols will not be found after this pr, such as DebugInfoWriter and at::cuda::CUDAEvent

**Pull request closed without merging**